docs/log-correlation.asciidoc

```diff
@@ -0,0 +1,172 @@
+- NLog
+
+If your favorite logging framework is not already supported, there are two other options:
+
+* Open a feature request, or contribute code, for additional support, as described in https://github.com/elastic/apm-agent-dotnet/blob/master/CONTRIBUTING.md[CONTRIBUTING.md].
+* Manually inject trace IDs into log records, as described below in <<log-correlation-manual>>.
+
+Regardless of how you integrate APM with logging, you can use {filebeat-ref}[Filebeat] to
+send your logs to Elasticsearch, in order to correlate your traces and logs and link from
+APM to the {apm-app-ref}/xpack-logs.html[Logs app].
+
+[[serilog]]
+== Serilog
+
+We offer a https://github.com/serilog/serilog/wiki/Enrichment[Serilog Enricher] that adds the trace id to every log line that is created during an active trace.
+
+The enricher lives in the https://www.nuget.org/packages/Elastic.Apm.SerilogEnricher[Elastic.Apm.SerilogEnricher] NuGet package.
+
+You can enable it when you configure your Serilog logger:
+
+```
+var logger = new LoggerConfiguration()
+   .Enrich.WithElasticApmCorrelationInfo()
+   .WriteTo.Console(outputTemplate: "[{ElasticApmTraceId} {ElasticApmTransactionId} {Message:lj} {NewLine}{Exception}")
+   .CreateLogger();
+```
+
+In the code snippet above `.Enrich.WithElasticApmCorrelationInfo()` enables the enricher, which will set 2 properties for log lines that are created during a transaction:
+
+- ElasticApmTransactionId
+- ElasticApmTraceId
+
+As you can see, in the `outputTemplate` of the Console sink these two properties are printed. Of course they can be used with any other sink.
+
+If you want to send your logs directly to Elasticsearch you can use the https://www.nuget.org/packages/Serilog.Sinks.Elasticsearch[Serilog.Sinks.ElasticSearch] package. Furthermore, you can pass the `EcsTextFormatter` from the   https://www.nuget.org/packages/Elastic.CommonSchema.Serilog[Elastic.CommonSchema.Serilog] package to the Elasticsearch sink, which formats all your logs according to Elastic Common Schema (ECS) and it makes sure that the trace id ends up in the correct field.
+
+Once you added the two packages mentioned above, you can configure your logger like this:
+
+```
+Log.Logger = new LoggerConfiguration()
+.Enrich.WithElasticApmCorrelationInfo()
+.WriteTo.Elasticsearch(new ElasticsearchSinkOptions(new Uri("http://localhost:9200"))
+{
+  CustomFormatter = new EcsTextFormatter()
+})
+.CreateLogger();
+``` 
+
+With this setup the application will send all the logs automatically to Elasticsearch and you will be able to jump from traces to logs and from logs to traces.
+
+
+[[nlog]]
+== NLog
+
+For NLog, we offer two https://github.com/NLog/NLog/wiki/Layout-Renderers[LayoutRenderers] that inject the current trace and transaction id into logs. 
+
+In order to use them, you need to add the https://www.nuget.org/packages/Elastic.Apm.NLog[Elastic.Apm.NLog] NuGet package to your application and load it in the `<extensions>` section of your NLog config file:
+
+```
```

Comment:
Let's change this from a large code block to a structured code block so it renders in the same format as the other code on the page. The backticks are pretty ugly:
<img width="762" alt="Screen Shot 2019-12-18 at 11 29 56 AM" src="https://user-images.githubusercontent.com/5618806/71116872-c5bdae00-2189-11ea-9765-b221313aabf9.png">


```suggestion
[source,xml]
----
```